pkg/blobcache/blobcache.go

```diff
@@ -63,6 +63,8 @@ type blobCacheSource struct {
 	cacheHits   int64
 	cacheMisses int64
 	cacheErrors int64
+	// this mutex synchronizes the counters above
+	mu sync.Mutex
```

Comment:
(Extreme nit: I’d prefer having the mutex above the protected data, or comments on each of the counter lines, to make this even harder to overlook.)